### ISSUE TITLE
no-redeclare: Errors on local globals definition overriding default globals definition

### ISSUE THREAD

**Issue by JHilker** [eslint/eslint#12534]
**Tell us about your environment**

```
Environment Info:

Node version: v12.10.0
npm version: v6.10.3
Local ESLint version: v6.5.1
Global ESLint version: Not found
```

**What parser (default, Babel-ESLint, etc.) are you using?**

Occurs with both default and `babel-eslint@10.0.3`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: ['eslint:recommended'],

  globals: {
    foo: 'readonly'
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* globals foo:writable */

foo = false;

console.log(foo);
```

```bash
yarn eslint foo.js
```

**What did you expect to happen?**

`no-redeclare` Should allow you to override the globals definition of `foo` from `readonly` to `writable` for the context of this file without erroring.

**What actually happened? Please include the actual, raw output from ESLint.**

`no-redeclare` flags `foo` as a redeclaration, even though it's to a new value. Technically it's still a redeclaration, but I would expect the local config to take precedence. Disabling either the globals line or the line with the assignment is required currently.

```
➜  some-repo git: ✗ yarn eslint foo.js
yarn run v1.17.3
$ /some-repo/node_modules/.bin/eslint foo.js

/some-repo/foo.js
  1:11  error  'foo' is already defined as a built-in global variable  no-redeclare

✖ 1 problem (1 error, 0 warnings)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command
```

Connected to https://eslint.org/docs/user-guide/migrating-to-6.0.0#-the-no-redeclare-rule-is-now-more-strict-by-default

**Are you willing to submit a pull request to fix this bug?**

With some insight on where to best make the fix, sure. It looks like all global comments are iterated regardless of the `eslintImplicitGlobalSetting`: https://github.com/eslint/eslint/blob/586855060afb3201f4752be8820dc85703b523a6/lib/rules/no-redeclare.js#L73

I suspect in the case of both a declared global and a global comment, we'd want to avoid yielding the builtin iff the last comment had a different `eslintImplicitGlobalSetting` from the builtin?

https://github.com/eslint/eslint/blob/586855060afb3201f4752be8820dc85703b523a6/lib/rules/no-redeclare.js#L61-L66

**Comment by mysticatea** [comment]
Thank you for your report.

This is intentional behavior.

Because the change from `readable` to `writable` is similar to the following.

```
const foo = 0 // readable
let foo // try to change to writable
```

Such a redeclaration causes errors. If you really want such a redeclaration, you can use `eslint-disable` directive comments.

Thanks.

**Comment by JHilker** [comment]
Ahh yeah, that's a good way to think about it 👍. I've always considered it `readonly` / `writeable` for globals in terms of what *should* be done rather than what *could* be done, but if this is WAD, happy to use `eslint-disable` comments instead. Thanks!
